man/src/network/create.md

```diff
@@ -134,3 +134,44 @@ $ docker network create -d overlay \
   --opt encrypted=true \
   my-ingress-network
 ```
+
+### Run services on predefined networks
+
+You can now create services on the predefined docker networks `bridge` and `host`.
+
+```bash
+$ docker service create --name my-service \
+  --network host \
+  --replicas 2 \
+  busybox top
+```
+
+### Swarm networks with local scope drivers
+
+You can now create a swarm network with local scope network drivers. You do so
+by promoting the network scope to `swarm` during the creation of the network. 
+You will then be able to use this network when creating services. 
+
+```bash
+$ docker network create -d bridge \
+  --scope swarm \
+  --attachable \
+  swarm-network
+```
+
+For network drivers which provide connectivity across hosts (ex. macvlan), if
+node specific configurations are needed in orther to plumb the network on each
+host, you will supply that configuration via a configuration only network.
+When you create the swarm scoped network, you will then specify the name of the 
+network which contains the configuration.
+
+
+```bash
+node1$ docker network create --config-only --subnet 192.168.100.0/24 --gateway 192.168.100.115 mv-config
```

Comment:
Let me think out loud; The problem we're trying to solve is that;

we want to be able to use (e.g.) macvlan networks in a swarm, however;

- those networks require (may require) a different configuration per node
- those differences in configuration cannot easily be defined in a single `docker network create` call, because;
  - nodes can be added/removed after the network is created
  - we would have to define a "range" of options for each node to "pick" from
  - we can't specify them in the `docker service create` call, because we don't know which nodes a task will land on
- we don't want to create the network on each node up-front, because we only want it to be created the moment a task lands on that node, and want it to be removed when the task (all tasks using that network) is destroyed

the problem I see right now is that;

- (currently) you have to go to each node to create a `--config-only` network, before the node can participate in such a network
- looking at the options in the examples, `--config-only` configurations don't seem to be really "reusable" (there may be examples where it is of course), so basically for _each_ network I'll create using a `config`, I will be creating a separate config (so there will be a 1-on-1 mapping between "configs" and "networks")
- confusing UX (see the above discussion)


For _overlay_ networks, basically the same is done currently, only the "config" (/"defaults") can be automatically determined by swarm/libnetwork, so all this takes place behind the scenes. In addition, those "configs" are not visible to the user as "config", the network just shows up on a node once it's being rolled out / extended to the node.

Allowing services to use macvlan and local (bridge) networks is something often requested; Would it be an acceptable start to split the "config" feature from that feature, and require the user to create the network up-front on each node, before deploying a service? i.e., to use a macvlan network;

- go to each node and create the network (question: I assume the network needs to have the same `ID` on each node, or is it sufficient to have the same name? what happens if `node A` has a bridge network `"foo"`, but `node B` has a macvlan network named `"foo"`?)
- after the networks are created, deploy the service
- will this be a problem for `worker` nodes? i.e. can a `worker` node create a network, and have `swarmkit` know about its existence?


Thinking about that last bullet; `docker network create --config` can be run on _any_ node. So far we had a "least privilege" model in swarm; do we cross that boundary here? (perhaps we already did, with "volumes" that can be created on worker nodes)